app/models/manageiq/providers/openshift/container_manager.rb

```diff
@@ -37,11 +37,11 @@ def self.default_port
   end
 
   def v3?
-    api_version.to_s =~ /3\..*/
+    api_version.to_s.first == "3"
   end
 
   def v4?
-    api_version.to_s =~ /4\..*/
+    api_version.to_s.first == "4"
```

Comment:
I copied this logic from the VMwareWebService VimService code which does something similar but it didn't work as I was expecting :(  Added specs so this doesn't happen again